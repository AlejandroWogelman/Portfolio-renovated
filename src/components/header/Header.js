import React from "react";
import "./header.css";

export const Header = () => {
  return (
    <header className="header">
      <nav>
        <ul>
          <li>
            <a href="">About</a>
          </li>
          <li>
            <a href="">Work</a>
          </li>
          <li>
            <a href="">Contact</a>
          </li>
        </ul>
      </nav>
      <div className="header-content">
        <div className="first-box">
          <div className="two-box">
            <h2>Hola!</h2>
            <h1>
              Soy Alejandro, <br />
              <span>Frontend developer</span>
            </h1>
          </div>
        </div>
        <img
          src={require("../../assets/code2.png").default}
          alt="coding"
          width="100%"
          height="100%"
        />
      </div>
    </header>
  );
};
