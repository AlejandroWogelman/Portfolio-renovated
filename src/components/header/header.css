.header {
  height: 100vh;
  background-color: rgb(62, 182, 187);
  position: relative;
  color: aliceblue;
}
nav {
  padding: 1rem 0;
  width: 95%;
  max-width: 1440px;
  position: fixed;
  background-color: rgb(62, 182, 187);
  z-index: 1;
  border-radius: 0 0 5px 5px;

  left: 0;
  right: 0;

  margin: auto;
}

ul {
  display: flex;
  justify-content: center;
}
ul li {
  margin: 10px 2rem;
  list-style: none;
}
ul li a {
  text-decoration: none;
  color: aliceblue;
  padding: 10px;
  transition: 0.5s;
}
ul li a:hover {
  background-color: rgb(33, 142, 146);
}
.header-content {
  display: flex;
  height: 90%;
  width: 100%;
  justify-content: center;
  align-items: center;
}
.first-box {
  height: 160px;
  width: 100%;
  background-color: azure;
  width: min-content;
  border: dotted rgba(0, 0, 0, 0.472) 5px;
  display: flex;
  margin-right: 5%;
  transform: translateY(-20%);
}
.two-box {
  padding: 20px 30px;
  transform: translateY(-20%) translateX(20%);
  border: 5px rgb(99, 192, 197) solid;
  background-color: rgb(62, 182, 187);
  box-shadow: rgba(0, 0, 0, 0.164) -8px 8px 1px;
}
.two-box h2 {
  font-family: "Reem Kufi", sans-serif;
  font-size: 6rem;
  transform: translateY(-10px);
}
.two-box h1 {
  margin-top: 80px;
  transform: translateX(-20%);
  font-size: 2rem;
}
.two-box h1 span {
  font-size: 1.2rem;
}
.header-content img {
  margin-left: 5%;
  height: 250px;
  width: 250px;
}
@media screen and (max-width: 720px) {
  .header-content img {
    display: none;
  }
  .two-box {
    padding: 10px 15px;
  }
  .two-box h2 {
    font-size: 5.5rem;
  }
  .first-box {
    margin-right: 12%;
  }
}
@media screen and (max-width: 500px) {
  ul li {
    margin: 10px 1rem;
  }
}
