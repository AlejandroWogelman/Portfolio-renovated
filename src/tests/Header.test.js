import "@testing-library/jest-dom";
import { Header } from "../components/header/Header";
import { shallow } from "enzyme";

describe("<Header/>", () => {
  const wrapper = shallow(<Header />);

  test("Render of header component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Correct anchor navigation", () => {
    const about = wrapper.find("a").at(0);
    const work = wrapper.find("a").at(1);
    const contact = wrapper.find("a").at(2);

    expect(about.text()).toBe("About");
    expect(about.prop("href")).toBe("#about");

    expect(work.text()).toBe("Work");
    expect(work.prop("href")).toBe("#work");

    expect(contact.text()).toBe("Contact");
    expect(contact.prop("href")).toBe("#contact");
  });
});
