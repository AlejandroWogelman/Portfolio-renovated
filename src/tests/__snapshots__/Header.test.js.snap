// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Render of header component 1`] = `
<header
  className="header"
>
  <nav
    className={null}
  >
    <ul>
      <li>
        <a
          href="#about"
        >
          About
        </a>
      </li>
      <li>
        <a
          href="#work"
        >
          Work
        </a>
      </li>
      <li>
        <a
          href="#contact"
        >
          Contact
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="header-content"
  >
    <div
      className="first-box"
    >
      <div
        className="two-box"
      >
        <h2>
          Hola!
        </h2>
        <h1>
          Soy Alejandro, 
          <br />
          <span>
            Frontend developer
          </span>
        </h1>
      </div>
    </div>
    <img
      alt="coding"
      height="100%"
      width="100%"
    />
  </div>
</header>
`;
